dsc/tests/dsc_settings.tests.ps1

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'tests for dsc settings' {
+    BeforeAll {
+
+        $script:policyFilePath = if ($IsWindows) {
+            Join-Path $env:ProgramData "dsc" "settings.dsc.json"
+        } else {
+            "/etc/.dsc/settings.dsc.json"
+        }
+
+        $script:dscHome = (Get-Command dsc).Path | Split-Path
+        $script:dscSettingsFilePath = Join-Path $script:dscHome "settings.dsc.json"
+        $script:dscDefaultv1SettingsFilePath = Join-Path $script:dscHome "default_settings.v1.dsc.json"
+        $script:dscDefaultv1SettingsJson = Get-Content -Raw -Path $script:dscDefaultv1SettingsFilePath
+
+        if ($IsWindows) { #"Setting policy on Linux requires sudo"
+            $script:policyDirPath = $script:policyFilePath | Split-Path
+            New-Item -ItemType Directory -Path $script:policyDirPath | Out-Null
+        }
+
+        #create backups of settings files
+        $script:dscSettingsFilePath_backup = Join-Path $script:dscHome "settings.dsc.json.backup"
+        $script:dscDefaultv1SettingsFilePath_backup = Join-Path $script:dscHome "default_settings.v1.dsc.json.backup"
+        Copy-Item -Force -Path $script:dscSettingsFilePath -Destination $script:dscSettingsFilePath_backup
+        Copy-Item -Force -Path $script:dscDefaultv1SettingsFilePath -Destination $script:dscDefaultv1SettingsFilePath_backup
+    }
+
+    AfterAll {
+        Remove-Item -Force -Path $script:dscSettingsFilePath_backup
+        Remove-Item -Force -Path $script:dscDefaultv1SettingsFilePath_backup
+        if ($IsWindows) { #"Setting policy on Linux requires sudo"
+            Remove-Item -Recurse -Force -Path $script:policyDirPath
+        }
+    }
+
+    AfterEach {
+        Copy-Item -Force -Path $script:dscSettingsFilePath_backup -Destination $script:dscSettingsFilePath
+        Copy-Item -Force -Path $script:dscDefaultv1SettingsFilePath_backup -Destination $script:dscDefaultv1SettingsFilePath
+        if ($IsWindows) { #"Setting policy on Linux requires sudo"
+            Remove-Item -Path $script:policyFilePath -ErrorAction SilentlyContinue
+        }
+    }
+
+    It 'ensure a new tracing value in settings has effect' {
+        
+        $script:dscDefaultv1SettingsJson.Replace('"level": "WARN"', '"level": "TRACE"') | Set-Content -Force -Path $script:dscSettingsFilePath
+
+        dsc resource list 2> $TestDrive/tracing.txt
+        "$TestDrive/tracing.txt" | Should -FileContentMatchExactly "Trace-level is Trace"
+    }
+
+    It 'ensure a new resource_path value in settings has effect' {
+        
+        $script:dscDefaultv1SettingsJson.Replace('"directories": []', '"directories": ["TestDir"]') | Set-Content -Force -Path $script:dscSettingsFilePath
```

Comment:
Would be better to have at least two paths

Reply:
Updated